development/generate-beta-commit.js

```diff
@@ -0,0 +1,30 @@
+const { promisify } = require('util');
+const exec = promisify(require('child_process').exec);
+const VERSION = require('../package.json').version;
+
+start().catch(console.error);
+
+async function start() {
+  let betaVersion;
+  if (VERSION.includes('beta')) {
+    const splitVersion = VERSION.split('-beta.');
+    const currentBetaVersion = Number(splitVersion[1]) + 1;
+    betaVersion = `${splitVersion[0]}-beta.${currentBetaVersion}`;
+    // Remove auto generated stableVersion an , after packageManager line to achieve bump
+    // You can find the issue here: https://github.com/yarnpkg/berry/issues/4328
+    await exec(
```

Comment:
The syntax of the sed commands and options can vary slightly between different platforms. I think `-i`, particularly, is not compatible with Windows.

Can you consider reading `package.json`, performing the operation on the in-memory JSON object, and then writing `package.json` using node standard packages? This would ensure cross-platform compatibility.

Reply:
there's some permission problem i encountered when reading / modifying package.json file.

So i was reading the solution in this issue, https://github.com/yarnpkg/berry/issues/4328, 
and using the script to manipulate the package.json directly,
and i got permission denied. 
```
#!/bin/bash

# Remove `stableVersion` before releasing, as it's buggy.
# https://github.com/yarnpkg/berry/issues/3868
echo "$(jq 'del(.stableVersion)' package.json)" >package.json
```

what if i use `awk` instead to perform?

Reply:
got some help in person and use fs module now . it is more clear now, thank you !